tests/scattering2d/test_torch_scattering2d.py

```diff
@@ -302,52 +298,101 @@ def test_device_mismatch(self, backend_device):
 class TestFFT:
     @pytest.mark.parametrize('backend', backends)
     def test_fft(self, backend):
-        x = torch.randn(2, 2, 2)
+        import numpy as np
+        
+        def coefficent(n):
+            return np.exp(-2*np.pi*1j*n)
 
-        y = torch.empty_like(x)
-        y[0, 0, :] = x[0, 0, :] + x[0, 1, :] + x[1, 0, :] + x[1, 1, :]
-        y[0, 1, :] = x[0, 0, :] - x[0, 1, :] + x[1, 0, :] - x[1, 1, :]
-        y[1, 0, :] = x[0, 0, :] + x[0, 1, :] - x[1, 0, :] - x[1, 1, :]
-        y[1, 1, :] = x[0, 0, :] - x[0, 1, :] - x[1, 0, :] + x[1, 1, :]
+        x_r = np.random.rand(4, 4)
+        x_i = np.random.rand(4, 4)
 
-        z = backend.fft(x, direction='C2C')
+        x = x_r + 1j*x_i
 
-        assert torch.allclose(y, z)
+        I, J, K, L = np.meshgrid(np.arange(4), np.arange(4), np.arange(4),
+                np.arange(4), indexing='ij')
+        coefficents = coefficent(K * I / x.shape[0] + L * J / x.shape[1])
+
+        y = np.zeros(x.shape).astype('complex128')
+        
+        for k in range(4):
+            for l in range(4):
+                y[k, l] = (x*coefficents[..., k, l]).sum()
+
+        y_r = np.zeros(x.shape).astype('complex128')
+        
+        for k in range(4):
+            for l in range(4):
+                y_r[k, l] = (x_r*coefficents[..., k, l]).sum()
 
-        z = backend.fft(x, direction='C2C', inverse=True)
 
-        z = z * 4.0
+        y = torch.from_numpy(np.stack((y.real, y.imag), axis=-1))
+        y_r = torch.from_numpy(np.stack((y_r.real, y_r.imag), axis=-1))
+        x = torch.from_numpy(np.stack((x.real, x.imag), axis=-1))
+        x_r = torch.from_numpy(x_r).unsqueeze(-1)
+
+        z = torch.fft(x, 2)        
```

Comment:
Why are we testing the Torch FFT here? Shouldn't the test for `backend.rfft` below be enough?

Reply:
we don't have to, I wanted to just compute the fft so I could perform the ifft on `z`.

Reply:
Ok but can't we just use the `rfft` for that?

Reply:
yeah